src/Materials/Textures/texture.ts

```diff
@@ -456,7 +466,11 @@ export class Texture extends BaseTexture {
             this.vScale === this._cachedVScale &&
             this.uAng === this._cachedUAng &&
             this.vAng === this._cachedVAng &&
-            this.wAng === this._cachedWAng) {
+            this.wAng === this._cachedWAng &&
+            this.uRotationCenter === this._cachedURotationCenter &&
+            this.vRotationCenter === this._cachedVRotationCenter &&
+            this.wRotationCenter === this._cachedWRotationCenter &&
+            this.homogeneousRotationInUVTransform === this._cachedHomogeneousRotationInUVTransform) {

```

Comment:
I would vote for testing the rotationCenter only if homogeanous is true (that could save 3 tests for each texture matrix per frame)

Reply:
Those variables are also used in the "legacy" case and to me it appears it was a bug in the previous code: updating any of these values would not update the matrix whereas it should do.